plugins/woocommerce/tests/legacy/unit-tests/util/install.php

```diff
@@ -117,10 +117,16 @@ public function test_create_roles() {
 	 * Test - remove roles.
 	 */
 	public function test_remove_roles() {
-		WC_Install::remove_roles();
-
-		$this->assertNull( get_role( 'customer' ) );
-		$this->assertNull( get_role( 'shop_manager' ) );
+		try {
+			WC_Install::remove_roles();
+
+			$this->assertNull( get_role( 'customer' ) );
+			$this->assertNull( get_role( 'shop_manager' ) );
+		} finally {
+			// Always restore roles for subsequent tests.
```

Comment:
Given all the tests in this class deal with installation, maybe instead of trying to restore the state in this particular test we instead do it inside `tearDownAfterClass()` at the class level? As in we fully uninstall WC and reinstall via `WC_Install::install()`?

Not sure if re-installing surfaces other issues, but I do worry that `test_create_roles()` is also not restoring things to the previous state, so we're technically not running a "recently installed" WC anymore after that test runs.

If this is too much trouble, we can definitely go with the current approach.